pkg/dwerrors/errors.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2019-2023 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dwerrors
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/devfile/devworkspace-operator/pkg/provision/sync"
+)
+
+type RetryError struct {
+	// Err is the underlying error causing the problem. If nil, it is not included in the output of Error()
+	Err error
+	// Message is a user-friendly string explaining why the error occurred
+	Message string
+	// RequeueAfter represents how long we should wait before checking if storage is ready
```

Comment:
The documentation for `RequeueAfter` mentions storage, though RetryError is used for other things (e.g. the routing can produce this type of error), so this should probably be changed to be a bit more general